common/src/main/java/com/wynntils/models/character/CharacterModel.java

```diff
@@ -192,36 +190,37 @@ public void onContainerClick(ContainerClickEvent e) {
         }
     }
 
-    private void scanCharacterInfoPage() {
-        ScriptedContainerQuery query = ScriptedContainerQuery.builder("Character Info Query")
-                .onError(msg -> WynntilsMod.warn("Error querying Character Info: " + msg))
-
-                // Open compass/character menu
-                .then(QueryStep.useItemInHotbar(InventoryUtils.COMPASS_SLOT_NUM)
-                        .expectContainerTitle(ContainerModel.CHARACTER_INFO_NAME)
-                        .processIncomingContainer(this::parseCharacterContainer))
-                .build();
-
-        query.executeQuery();
-    }
-
-    private void scanSilverbullSubscriptionItem() {
-        ScriptedContainerQuery query = ScriptedContainerQuery.builder("Silverbull Subscription Query")
-                .onError(msg -> WynntilsMod.warn("Error querying Silverbull subscription: " + msg))
-
-                // Open /use menu
-                .then(QueryStep.sendCommand("use")
-                        .expectContainerTitle(ContainerModel.COSMETICS_MENU_NAME)
-                        .processIncomingContainer(this::parseCratesBombsCosmeticsContainer))
-                .build();
+    private void scanCharacterInfo() {
+        QueryBuilder queryBuilder = ScriptedContainerQuery.builder("Character Info Query");
+        queryBuilder.onError(msg -> WynntilsMod.warn("Error querying Character Info: " + msg));
+        // Open compass/character menu
+        queryBuilder.then(QueryStep.useItemInHotbar(InventoryUtils.COMPASS_SLOT_NUM)
+                .expectContainerTitle(ContainerModel.CHARACTER_INFO_NAME)
+                .processIncomingContainer(this::parseCharacterContainer));
+        if (System.currentTimeMillis() > silverbullExpiresAt.get()) {
+            // Open Cosmetics Menu
+            queryBuilder
+                    .then(QueryStep.clickOnSlot(COSMETICS_SLOT)
+                            .expectContainerTitle(ContainerModel.COSMETICS_MENU_NAME)
+                            .processIncomingContainer(this::parseCratesBombsCosmeticsContainer))
+                    .then(QueryStep.clickOnSlot(COSMETICS_BACK_SLOT)
+                            .expectContainerTitle(ContainerModel.CHARACTER_INFO_NAME));
+        } else {
+            WynntilsMod.info("Skipping silverbull subscription query ("
+                    + (silverbullExpiresAt.get() - System.currentTimeMillis()) + " ms left)");
+        }
+        // Open Guild Menu
+        queryBuilder.then(QueryStep.clickOnSlot(GUILD_MENU_SLOT)
+                .expectContainerTitle(ContainerModel.GUILD_MENU_NAME)
+                .processIncomingContainer(Models.Guild::parseGuildContainer));
```

Comment:
Can you click on the guild item, if you are not in a guild? Can you "expect" a container?

Reply:
This is something not actually tested by me, as I don't have an alt account nor want to leave my guild.
But my assumption is, that it would "fail" the parsing when the container isn't presented, with everything else already parsed at this point.

Reply:
This is the item for no guild and clicking it does nothing
![image](https://github.com/Wynntils/Artemis/assets/8337467/4f42800f-9b95-468d-a836-3c29c32fb8c2)

Reply:
Making this work without expected failure is a blocking change required for this PR. The query system should have the necessary conditional query steps that allow you to fix this.